frankenstein/ingester_lifecycle.go

```diff
@@ -30,23 +32,59 @@ const (
 	infName = "eth0"
 )
 
-// WriteIngesterConfigToConsul writes ingester config to Consul
-func WriteIngesterConfigToConsul(consulClient ConsulClient, listenPort int, numTokens int) error {
-	log.Info("Adding ingester to consul")
+// IngesterRegistration is the registration of an ingester with a register.
+type IngesterRegistration interface {
+	Unregister() error
+}
```

Comment:
(I'm not a big stickler for things like this, btw..)

"A great rule of thumb for Go is accept interfaces, return structs."

http://dave.cheney.net/2016/08/20/solid-go-design about 2/3 way down.